src/Symfony/Component/Validator/Mapping/Factory/LazyLoadingMetadataFactory.php

```diff
@@ -117,35 +117,24 @@ public function getMetadataFor($value)
 
     private function mergeConstraints(ClassMetadata $metadata)
     {
-        // Include constraints from the parent class
-        if ($parent = $metadata->getReflectionClass()->getParentClass()) {
-            $metadata->mergeConstraints($this->getMetadataFor($parent->name));
-        }
-
-        $interfaces = $metadata->getReflectionClass()->getInterfaces();
-
-        $interfaces = array_filter($interfaces, function ($interface) use ($parent, $interfaces) {
-            $interfaceName = $interface->getName();
-
-            if ($parent && $parent->implementsInterface($interfaceName)) {
-                return false;
+        if (!$metadata->getReflectionClass()->isInterface()) {
```

Comment:
I am not convinced that this will work in all cases. What happens now when a class implements an interface that extends another interface (i.e. will the test added still pass if `PropertyGetter` doesn't explicitly implement the `ChildGetterInterface`?

Reply:
`ReflectionClass::getInterfaces` returns all interfaces implemented by the class.

The main idea here is to exclude the interfaces implemented by the parent class when processing the class itself.

Thus, if class `A` extends class` B` and implements the interfaces `H`,` I` and `J` (the hierarchy of interfaces does not make sense here), and class` B` implements `I` and` J`, there is a sequence:
- go to class `B` and add constraints from the interfaces `I` and `J`, and from `B`;
- add constraints from the interface `H` (`I` and `J` are implemented by class `B` and are skipped), and from `A`.

Reply:
Sounds reasonable, can we use early returns here to reduce the diff?

Like this:

```php
private function mergeConstraints(ClassMetadata $metadata)
{
    if ($metadata->getReflectionClass()->isInterface()) {
        return;
    }

    // ...
}
```

Reply:
Done